### PR TITLE
Nats-Server: Add version 2.9.0

### LINKED ISSUE #3953: [Request] add Nats-Server
### Criteria

For a package to be acceptable in the Main bucket, it should be:

- [x] a non-GUI tool
- [x] a reasonably well-known and widely used developer tool (e.g. if it's a GitHub project, it should have at least 500 stars and 150 forks)
- [x] the latest stable version of the program
- [x] the full version i.e. not a trial version
- [x] a fairly standard install (e.g. uses a version-specific download URL, no elaborate pre/post install scripts)

### Information

**Name:** nats-server

**Description:**  A server for NATS, the cloud and edge native messaging system supporting lots of different features for communication and messaging.

**Homepage:** https://github.com/nats-io/nats-server

**Download link(s):** https://github.com/nats-io/nats-server/releases/

**Some indication of popularity/repute:** 1.1k forks, 11.5k stars. https://nats.io/ list of big companies using NATS.

### PR DESCRIPTION
**Pull request by Lemorz56**
Closes #3953

- [X] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

I would like to hold on with merging the PR until I've gotten a final response from the NATS team. 
Still like to get feedback and verify it.

### CODE REVIEW

**Review by rashil2000 — CHANGES_REQUESTED** [review]
Please update the manifest to version 2.9.1

File: bucket/nats-server.json
Comment (Lemorz56) [reply bucket/nats-server.json]: Fixed :)

**Comment by rashil2000** [comment]
Couple of more things:
- Add 32bit variant: https://github.com/nats-io/nats-server/releases/download/v2.9.1/nats-server-v2.9.1-windows-386.zip
- Add autoupdate hash extraction: https://github.com/nats-io/nats-server/releases/download/v2.9.1/SHA256SUMS

**Comment by Lemorz56** [comment]
> Couple of more things:
> 
> * Add 32bit variant: https://github.com/nats-io/nats-server/releases/download/v2.9.1/nats-server-v2.9.1-windows-386.zip
> * Add autoupdate hash extraction: https://github.com/nats-io/nats-server/releases/download/v2.9.1/SHA256SUMS

Sorry for taking so long. Added the 32bit exe and the hash for versioning. I'm not sure the regex for the hash works as intended though so I'd love if you could check it out. Seems to work locally though 👍🏼

File: bucket/nats-server.json
Comment (Lemorz56) [inline bucket/nats-server.json:33]: I have a feeling that 64 shouldn't be there at the end?

Comment (rashil2000) [reply bucket/nats-server.json:33]: The `find` field can be removed here entirely

Comment (rashil2000) [reply bucket/nats-server.json:33]: Also, you should update the manifest to 2.9.2 now

Comment (Lemorz56) [reply bucket/nats-server.json]: I've removed the find field. And I have already changed it to be 2.9.2

**Comment by Lemorz56** [comment]
I'm still awaiting an answer from the NATS core team. I'll poke them before you/me merge.

**Comment by scottf** [comment]
I'm a committer on NATS team. We approve this PR.

**Comment by rashil2000** [comment]
Only members of the ScoopInstaller 'maintainers' team can approve/modify changes here.

**Comment by rashil2000** [comment]
Another thing: Scoop just released support for Arm64 architecture. @Lemorz56 could you add that too?

**Comment by Lemorz56** [comment]
Added arm64 and bumped version to 2.9.3 @rashil2000

**Comment by scottf** [comment]
LGTM

**Pull request merged**
